src/pages/admin/team/components/delete-team-category-modal/DeleteTeamCategoryModal.tsx

```diff
@@ -0,0 +1,106 @@
+import { useEffect, useState } from 'react';
+import { Modal } from '../../../../../components/common/modal/Modal';
+import { Button } from '../../../../../components/admin/button/Button';
+import { SingleSelectInputGroup } from '../../../../../components/admin/input-groups/single-select-input-group/SingleSelectInputGroup';
+import { HintBox } from '../../../../../components/admin/hint-box/HintBox';
+import { useAdminClient } from '../../../../../hooks/admin/use-admin-client/useAdminClient';
+import { COMMON_TEXT_ADMIN } from '../../../../../const/admin/common';
+import { TeamCategory } from '../../../../../types/admin/team-category';
+import { TEAM_CATEGORY_VALIDATION } from '../../../../../const/admin/team';
+import { TeamCategoriesApi } from '../../../../../services/api/admin/team/team-categories/team-categories-api';
+import './DeleteTeamCategoryModal.scss';
+
+interface DeleteTeamCategoryModalProps {
+    isOpen: boolean;
+    onClose: () => void;
+    onConfirm: (categoryId: number) => void;
+    categories: TeamCategory[];
+}
+
+export const DeleteTeamCategoryModal = ({ isOpen, onClose, onConfirm, categories }: DeleteTeamCategoryModalProps) => {
+    const client = useAdminClient();
+    const [selectedCategory, setSelectedCategory] = useState<TeamCategory>();
+    const [isSubmitting, setIsSubmitting] = useState(false);
+    const [error, setError] = useState('');
+
+    const handleConfirmation = async () => {
+        if (isSubmitting || !selectedCategory || selectedCategory.teamMembersCount > 0) return;
+
+        try {
+            setError('');
+            setIsSubmitting(true);
+
+            await TeamCategoriesApi.delete(client, selectedCategory.id);
+
+            onConfirm(selectedCategory.id);
+            setSelectedCategory(undefined);
+            onClose();
+        } catch {
+            setError(COMMON_TEXT_ADMIN.CATEGORIES.FORM.MESSAGE.FAIL_TO_DELETE_CATEGORY);
+        } finally {
+            setIsSubmitting(false);
+        }
+    };
+
+    const handleClose = () => {
+        if (isSubmitting) return;
+        setSelectedCategory(undefined);
+        onClose();
+    };
+
+    useEffect(() => {
+        if (isOpen) {
+            setError('');
+            if (categories.length > 0) {
+                setSelectedCategory(categories[0]);
+            }
+        }
+    }, [isOpen, categories]);
+
+    const handleCategoryChange = (category: TeamCategory) => {
+        setSelectedCategory(category);
+    };
+
+    return (
+        <Modal isOpen={isOpen} onClose={handleClose}>
+            <Modal.Title>{COMMON_TEXT_ADMIN.CATEGORIES.FORM.TITLE.DELETE_CATEGORY}</Modal.Title>
+            <Modal.Content>
+                <div className="program-form-main">
```

Comment:
Just suggestion to review CSS class names and come up with better naming, we're in `Delete` modal 
```suggestion
                <div className="delete-category-modal-content">
```

Reply:
Done